src/components/screens/SignTransactionDetails/hooks/useSignTransactionDetails.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  FeeBumpTransaction,
+  Operation,
+  Transaction,
+  TransactionBuilder,
+} from "@stellar/stellar-sdk";
+import { decodeMemo } from "components/screens/SignTransactionDetails/helpers/parseTransaction";
+import {
+  SignTransactionDetailsInterface,
+  SignTransactionSummaryInterface,
+} from "components/screens/SignTransactionDetails/types";
+import { mapNetworkToNetworkDetails, OPERATION_TYPES } from "config/constants";
+import { useAuthenticationStore } from "ducks/auth";
+import { stroopToXlm } from "helpers/formatAmount";
+
+interface UseSignTransactionDetailsParams {
+  xdr: string;
+}
+
+interface BuildSummaryParams {
+  transaction: Transaction | FeeBumpTransaction;
+  xdr: string;
+}
+
+const buildSummary = (
+  params: BuildSummaryParams,
+): SignTransactionSummaryInterface => {
+  const { transaction, xdr } = params;
+
+  let sequence = "";
+  let memo = {};
+
+  if (!("innerTransaction" in transaction)) {
+    sequence = transaction.sequence;
+    memo = transaction.memo;
+  }
+
+  const decodedMemo = decodeMemo(memo);
+  const operations = transaction.operations.map(
+    (op) => OPERATION_TYPES[op.type] || op.type,
+  );
+
+  const summary: SignTransactionSummaryInterface = {
+    operationsCount: operations.length,
+    feeXlm: stroopToXlm(String(transaction.fee)).toString(),
+    sequenceNumber: sequence,
+    memo: decodedMemo,
+    xdr,
+  };
+
+  return summary;
+};
+
+const buildAuthEntries = (transaction: Transaction | FeeBumpTransaction) => {
+  const hasAuthEntries = transaction.operations.some(
+    (op) => op.type === "invokeHostFunction" && op.auth && op.auth.length,
+  );
+
+  if (!hasAuthEntries) return [];
+
+  return (
+    (transaction.operations[0] as Operation.InvokeHostFunction).auth?.map(
```

Comment:
will the auth entries alway be on the first operation? it seems a little unexpected/inconsistent to use `transaction.operations.some()` on line 55 but then here we only look at the first operation

Reply:
Yeah, I think looping through all of them makes sense at a first, but I'm curious to see if @aristidesstaffieri has some input or good reason on why we should look only at the first operation so we avoid doing unnecessary loops